lib/managers/files.js

```diff
@@ -968,6 +991,230 @@ Files.prototype.deleteVersion = function(fileID, versionID, callback) {
+ * uploading pieces of the file.
+ *
+ * API Endpoint: '/files/:fileID/upload-session'
+ * Method: POST
+ *
+ * @param {string} fileID - The ID of the file to upload a new version of
+ * @param {int} size - The size of the file that will be uploaded
+ * @param {Function} callback - Passed the upload session info if successful
+ * @returns {void}
+ */
+Files.prototype.createNewVersionUploadSession = function(fileID, size, callback) {
+
+	var apiURL = this.client._uploadBaseURL + urlPath(BASE_PATH, fileID, UPLOAD_SESSION_SUBRESOURCE),
+		params = {
+			// @TODO: When API supports passing params via body, switch to body
+			qs: {
+				file_size: size
+			}
+		};
+
+	this.client.post(apiURL, params, this.client.defaultResponseHandler(callback));
+};
+
+/**
+ * Uploads a chunk of a file to an open upload session
+ *
+ * API Endpoint: '/files/upload-session/:sessionID'
+ * Method: PUT
+ *
+ * @param {string} sessionID - The ID of the upload session to upload to
+ * @param {Buffer|string} part - The chunk of the file to upload
+ * @param {int} offset - The byte position where the chunk begins in the file
+ * @param {int} totalSize - The total size of the file being uploaded
+ * @param {?Object} options - Optional parameters, can be left null in most cases
+ * @param {string} [options.partID] - 8-character hexadecimal string uniquely denoting this part upload request
+ * @param {Function} callback - Passed the part definition if successful
+ * @returns {void}
+ */
+Files.prototype.uploadPart = function(sessionID, part, offset, totalSize, options, callback) {
+
+	if (!options) {
+		options = {};
+	}
+
+	var apiURL = this.client._uploadBaseURL + urlPath(BASE_PATH, UPLOAD_SESSION_SUBRESOURCE, sessionID),
+		partID = options.partID || generatePartID(),
+		params = {
+			headers: {
+				'Content-Type': 'application/octet-stream',
+				Digest: 'SHA=' + crypto.createHash('sha1').update(part).digest('base64'),
+				'X-Box-Part-Id': partID,
+				'Content-Range': `bytes ${offset}-${offset + part.length - 1}/${totalSize}`
+			},
+			json: false,
+			body: part
+		};
+
+	this.client.put(apiURL, params, function(err, response) {
+
+		if (err) {
+			callback(err);
+			return;
+		}
+
+		if (response.statusCode !== 200) {
+			callback(errors.buildUnexpectedResponseError(response));
+			return;
+		}
+
+		// @TODO: When API passes back chunk object, return to user directly
+		callback(null, {
+			part_id: partID,
+			offset: offset,
+			size: part.length
+		});
+	});
+};
+
+/**
+ * Commit an upload session after all parts have been uploaded, creating the new file
+ *
+ * API Endpoint: '/files/upload-sessions/:sessionID/commit'
+ * Method: POST
+ *
+ * @param {string} sessionID - The ID of the upload session to commit
+ * @param {string} fileHash - The base64-encoded SHA-1 hash of the file being uploaded
+ * @param {UploadPart[]} parts - The list of uploaded chunks to be committed
+ * @param {?Object} options - Optional parameters, can be left null
```

Comment:
Since the param is in the middle of the list, you have to pass _something_, but either `undefined` or `null` is acceptable here if you don't want to pass any options.